adapters/sovrn/sovrn.go

```diff
@@ -150,6 +174,16 @@ func getTagid(sovrnExt openrtb_ext.ExtImpSovrn) string {
 	}
 }
 
+func getImpIdx(impId string, request *openrtb2.BidRequest) int {
+	for idx, imp := range request.Imp {
+		if imp.ID == impId {
+			return idx
+		}
+	}
+
+	return -1 // hopefully this never happens
```

Comment:
We should be returning an error if this does happen.. This would mean that for some reason your bidder bid for an imp that doesn't match an ID with the imps in the request. By returning an explicit error, it will be much easier to debug and fix the issue rather than silently just setting the bid type to `banner`

Reply:
I see your point. I believe I have fixed this correctly now. Please let me know if you feel otherwise.